Add pre-commit configuration

igorkruglyak opened the pull request:
Closes #12

Ref: https://pre-commit.com

This commit includes the initial configuration for
the following pre-commit hooks:
* no-commit-to-branch
* add-trailing-comma
* remove-tabs
* trailing-whitespace
* check-merge-conflict
* double-quote-string-fixer
* requirements-txt-fixer
* check-yaml
* check-docstring-first
* end-of-file-fixer
* name-tests-test
* check-ast
* debug-statements
* flake8
* pydocstyle
* pylint using "local" mode
* mypy

igorkruglyak committed:
Configure the pre-commit tool

Closes #12

Ref: https://pre-commit.com

This commit includes the initial configuration for
the following pre-commit hooks:
 *   no-commit-to-branch
 *   add-trailing-comma
 *   remove-tabs
 *   trailing-whitespace
 *   check-merge-conflict
 *   double-quote-string-fixer
 *   requirements-txt-fixer
 *   check-yaml
 *   check-docstring-first
 *   end-of-file-fixer
 *   name-tests-test
 *   check-ast
 *   debug-statements
 *   flake8
 *   pydocstyle
 *   pylint using "local" mode
 *   mypy

webknjaz reviewed (changes requested):
1) Link the issue using `Closes` keyword.
2) Fix the commit message and PR title style to match the recommendations you've learned from the articles.

igorkruglyak commented:
Closes

webknjaz commented:
Read  GitHub docs about keywords

igorkruglyak commented:
Closes #15

webknjaz commented:
Put that into PR description and a commit message.

igorkruglyak commented:
closes #15 
Fix PR

webknjaz commented:
That's a comment to the PR. I ask you to change the PR description.

igorkruglyak commented:
I do not understand what needs to be done. please help

webknjaz commented:
It's on top of this page.
<img width="414" alt="pr-description" src="https://user-images.githubusercontent.com/578543/59716092-9ac3c300-9215-11e9-8ef7-7491048809a5.png">

igorkruglyak commented:
change Add pre-commit configuration #15 ?

igorkruglyak commented:
that's right?

webknjaz commented:
Don't duplicate the PR title. The description should augment it if needed. Add more details.

webknjaz commented:
After fixing the PR, rewrite the message in this commit accordingly: https://github.com/kpi-web-guild/django-girls-blog-igorkruglyak/pull/15/commits/83b28abcd56d1a66bce106ad6f27d6c14cc77c09

webknjaz commented:
Hey @schenkh, would you mind helping out Ihor here?

webknjaz commented:
@igorkruglyak format your PR title in a nice readable way, using `*` for list items.

igorkruglyak commented:
what should i do next? how to do it: 'rewrite the message in this commit accordingly'

webknjaz commented:
#4 and #5 should help you. Also, use Google and StackOverflow. Search for interactive rebase and rewriting branches/commits, also force push.

igorkruglyak commented:
насправді я ще працюю над покращенням своєї англійської, поки що. ну, відповідно, не завжди все вірно розумію. голова вже кипить на елементарних речах. мені що, локально в гіті переписати в коміті повідомлення? запушити коміт по новому? що робити з цим пулл реквестом? Напиши українською, будь ласка, покроково, що саме треба виправити, а як його зробити не буде проблемою для мене.

webknjaz commented:
саме так і треба зробити. як перепишеш і запушиш гілку, то це відобразиться у ПР автоматично.

webknjaz commented:
Що гілку ти створював читачам комітів твоїх нецікаво. А ще, ти не встановлюєш пре-коміт цим комітом. Треба писати який ефект буде від застосування.

schenkh commented:
@webknjaz Здається Ігор вже розібрався. Або я ще якось можу допомогти?

webknjaz commented:
@schenkh із вмістом коміта: поясниш як покращити?

schenkh commented:
Переглянув .pre-commit-config.yaml і він, на мою думку, виглядає як потрібно. А за змістом коміта я припускаю що потрібно змінити "Install pre-commit" на "Add pre-commit configuration" чи "Configure pre-commit rules" щоб був більш зрозумілий сенс зроблених змін.

webknjaz commented:
а ще розширений опис коміта має непотрібні дані і погано форматований

schenkh commented:
Так, тепер бачу, коли клонував гілку. @igorkruglyak коли ми обговорювали опис коміту з @webknjaz я для себе зробив такий висновок що назва повинна бути "Що ви зробили в імперативному настрої" а тіло: "Поясніть, що і чому ви зробили - а не як..."

webknjaz commented:
іноді треба і "як", але високорівнево і лише тоді, коли з коміта не очевидно, наприклад, якщо щось неймовірно складне і заплутане.

igorkruglyak commented:
Дуже дякую вам обом за підказки.
Так піде?

Add pre-commit configuration

Install pre-commit for automatically identifying issues
Added .pre-commit-config.yaml config 
Added the following hooks:
 *   no-commit-to-branch
 *   add-trailing-comma
 *   remove-tabs
 *   trailing-whitespace
 *   check-merge-conflict
 *   double-quote-string-fixer
 *   requirements-txt-fixer
 *   check-yaml
 *   check-docstring-first
 *   end-of-file-fixer
 *   name-tests-test
 *   check-ast
 *   debug-statements
 *   flake8
 *   pydocstyle
 *   pylint using "local" mode
 *   mypy

webknjaz commented:
ні, ти нічого не інсталюєш. це неправда.

igorkruglyak commented:
Added?

webknjaz commented:
ти конфіг додаєш, не програму

igorkruglyak commented:
Може так 
Added .pre-commit-config.yaml for automatically identifying issues

webknjaz commented:
Я вже писав вище взагалі: не треба повторювати заголовок коміта в доповненому описі.

webknjaz commented:
Щось типу

```console
Configure the pre-commit tool

Closes #15

Ref: https://pre-commit.com

This commit includes the initial configuration for
the following pre-commit hooks:
* no-commit-to-branch
* add-trailing-comma
* remove-tabs
* trailing-whitespace
* check-merge-conflict
* double-quote-string-fixer
* requirements-txt-fixer
* check-yaml
* check-docstring-first
* end-of-file-fixer
* name-tests-test
* check-ast
* debug-statements
* flake8
* pydocstyle
* pylint using "local" mode
* mypy
```

igorkruglyak commented:
Дякую! Так справді набагато інформативніше.

webknjaz commented:
Це потім буде в дереві гіта і треба думати про те, щоб повідомлення було цінним для того, хто це читатиме в майбутньому. Це можеш бути ти або хтось інший.
Якщо правильно вести історію в гіті, то легко штукати коли проблеми виникли при необхідності. Також зрозуміло стає хід думок і послідовність змін. Стилістика теж важлива, варто слідувати практикам, якими послуговуються розробники самого гіта.

webknjaz commented:
Не забувай, що змінити й коміт треба, бо він теж потрапить до історії.

igorkruglyak commented:
Зрозумів. Дякую. Зараз зміню.

webknjaz commented:
> Closes #15

оце неправильно. має бути номер іш'ю, яке автоматично закриється, якщо змерджити цей ПР.

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,49 @@
+-   repo: https://github.com/pre-commit/mirrors-pylint
+    rev: v2.3.1
+    hooks:
+    - id: pylint
```
The task was to use a "local" mode for this linter. Here's an example: https://github.com/sanitizers/octomachinery/blob/93899d4/.pre-commit-config.yaml#L51-L58

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,53 @@
+    name: PyLint
+    files: \.py$
+    entry: python3 -m pylint.__main__
+args: []
```
This is broken.

igorkruglyak replied:
thanks

webknjaz reviewed (changes requested):
Are all checks green/passed when you run `python3.7 -m pre_commit` locally?

webknjaz commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,53 @@
+    hooks:
+    -   id: mypy
+
+- repo: local
```
Use the same indentation style for all parts of this YAML file.

igorkruglyak replied:
> So what's the output of `python3.7 -m pre_commit`?

Check Yaml...........................................(no files to check)Skipped
Debug Statements (Python)............................(no files to check)Skipped
Tests should end in _test.py.........................(no files to check)Skipped
Fix requirements.txt.................................(no files to check)Skipped
Trim Trailing Whitespace.............................(no files to check)Skipped
Fix End of Files.....................................(no files to check)Skipped
Check docstring is first.............................(no files to check)Skipped
Fix double quoted strings............................(no files to check)Skipped
Check python ast.....................................(no files to check)Skipped
Don't commit to branch...................................................Passed
Check for merge conflicts............................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
Add trailing commas..................................(no files to check)Skipped
CRLF end-lines checker...............................(no files to check)Skipped
CRLF end-lines remover...............................(no files to check)Skipped
No-tabs checker......................................(no files to check)Skipped
Tabs remover.........................................(no files to check)Skipped
Verifying PEP257 Compliance..........................(no files to check)Skipped
mypy.................................................(no files to check)Skipped
PyLint...............................................(no files to check)Skipped

webknjaz replied:
Reply to things where asked. It's a comment to the line 44.

webknjaz replied:
And this comment asks about another thing you ignored. See above.

igorkruglyak replied:
Мені не зовсім зрозуміло. Помістити цю відповідь нижче під тим питанням що виведе команда python3.7 -m pre_commit? Чи ще щось виправити?

webknjaz replied:
Перше повідомлення у треді до цього рядка про відступи, а твоя відповідь ні до чого тут

igorkruglyak replied:
Ну так, переплутав, не туди вліпив, вибач. Видалити? Щось маю зробити, щоб правильно було?

webknjaz replied:
Просто на майбутнє не плутай. Ну і ти от змінив відступи, а тут не позначив що вирішено. Кнопка для цього внизу є + можна написати, бо от мені сповіщення не приходять про пуші. Відповідно можеш довго чекати поки я сам згадаю перевірити.

webknjaz commented:
So what's the output of `python3.7 -m pre_commit`?

webknjaz commented:
Можеш мерджити це

igorkruglyak commented:
> Можеш мерджити це

Маєш на увазі локально в себе?

webknjaz commented:
можна кнопкою у цьому пул реквесті

igorkruglyak commented:
> Можеш мерджити це

Маєш на увазі локально в себе?

> можна кнопкою у цьому пул реквесті

Merging is blocked  - ось це тут пише в мене

webknjaz commented:
Можеш скриншот прислати?

webknjaz commented:
Налаштування трохи змінив, має бути зелена кнопка тепер.

igorkruglyak commented:
> Налаштування трохи змінив, має бути зелена кнопка тепер.

Так, тепер зелена.

webknjaz commented:
Тепер берися за інтеграцію трейвіса
